src/App.js

```diff
@@ -1,68 +1,89 @@
 import {nanoid} from 'nanoid';
-import {useState} from 'react';
+import {useEffect, useState} from 'react';
 import styled from 'styled-components';
 
 import CategoryButton from './components/CategoryButton';
-import Header from "./components/Header";
+import Header from './components/Header';
 import InputDataDialog from './components/InputDataDialog.js';
 import ReturnButton from './components/ReturnButton';
-import { initialInputData } from "./db";
-import { messages } from "./db";
+import {initialInputData} from './db';
+import {messages} from './db';
 
 export default function App() {
-  
   const [inputs, setInputs] = useState(initialInputData);
   const [showMessage, setShowMessage] = useState(messages[0].text);
   const [toggle, setToggle] = useState(true);
+  const [totalConsumption, setTotalConsumption] = useState({});
   const oldInputLength = initialInputData.length;
   const actualDate = new Date();
-  const formattedActualDate = actualDate.toLocaleDateString('en-GB', {year: 'numeric', month: '2-digit', day: '2-digit'})
-  const lastInputValue = inputs[0].value
+  const formattedActualDate = actualDate.toLocaleDateString('en-GB', {
+    year: 'numeric',
+    month: '2-digit',
+    day: '2-digit',
+  });
+
+  useEffect(() => {
+    function handleCalc() {
+      const total = inputs.map(input => input.increase).reduce((a, b) => a + b, 0);
+      const averageIncrease = total / inputs.length;
+      const averageIncreaseRounded = Math.round(averageIncrease);
+      setTotalConsumption({total, averageIncreaseRounded});
+    }
+    handleCalc();
+  }, [inputs]);
+
+  const indexLastInput = inputs.length;
+  const totalDate = actualDate - inputs[indexLastInput - 1].date;
```

Comment:
So here it would be:

```suggestion
  const totalDate = actualDate - inputs[indexLastInput].date;
```

Reply:
Another question is, if you really need totalDate ? I do not see you using it anywhere except of the console.log(). If so, it would be better to delete unnecessary code :)